File: src/main/java/com/ues/edu/modelo/Comprobante.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.ues.edu.modelo;

import jakarta.persistence.*;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;
import java.time.LocalDate;
import lombok.Data;
import lombok.NoArgsConstructor;
/**
 *
 * @author kikej
 */

@Data
@NoArgsConstructor
@Entity
@Table(name = "comprobantes")
public class Comprobante {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_comprobante")
    private Integer idComprobante;

    @NotNull(message = "El pago no puede ser nulo")
    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_pago", nullable = false, unique = true)
    private Pago pago;

    @NotBlank(message = "El número de comprobante no puede estar vacío")
    @Column(name = "numero_comprobante", nullable = false, length = 50, unique = true)
    private String numeroComprobante;

    @Column(name = "fecha_emision")
    private LocalDate fechaEmision;

    @NotNull(message = "El total no puede ser nulo")
    @DecimalMin(value = "0.0", inclusive = false, message = "El total debe ser mayor a cero")
    @Column(name = "total", nullable = false, precision = 10, scale = 2)
    private BigDecimal total;
}

File: src/main/webapp/js/comprobante.js
/* 
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/JSP_Servlet/JavaScript.js to edit this template
 */

function cargarComprobantes() {
    fetch("ComprobanteServlet")
        .then(res => res.json())
        .then(pintarTabla);
}

function pintarTabla(lista) {

    let tabla = `
    <table class="table table-bordered table-striped table-hover">
        <tr class="table-dark">
            <th>#</th>
            <th>Número de Comprobante</th>
            <th>Fecha de Emisión</th>
            <th>Total</th>
        </tr>`;

    lista.forEach(comprobante => {
        tabla += `
        <tr>
            <td>${comprobante.idComprobante}</td>
            <td>${comprobante.numeroComprobante}</td>
            <td>${comprobante.fechaEmision}</td>
            <td>$${comprobante.total}</td>
        </tr>`;
    });

    tabla += "</table>";

    $("#listaComprobantes").html(tabla);
}

cargarComprobantes();
